crypto/src/bls12381.rs

```diff
@@ -258,10 +261,57 @@ impl Display for BLS12381Signature {
     }
 }
 
+impl EncodeDecodeBase64 for BLS12381Signature {
+    fn encode_base64(&self) -> String {
+        base64ct::Base64::encode_string(self.as_bytes())
+    }
+
+    fn decode_base64(value: &str) -> Result<Self, eyre::Report> {
+        let bytes = base64ct::Base64::decode_vec(value).map_err(|e| eyre!("{}", e.to_string()))?;
+        BLS12381Signature::from_bytes(&bytes).map_err(|e| e.into())
+    }
+}
+
+impl Serialize for BLS12381Signature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            self.encode_base64().serialize(serializer)
+        } else {
+            let mut seq = serializer.serialize_tuple(BLS_SIGNATURE_LENGTH)?;
+            for element in self.sig.to_bytes() {
+                seq.serialize_element(&element)?;
+            }
+            seq.end()
```

Comment:
OK, so for both serialization and deserialization of bytes, you'll want to have a look at this explanation:
https://github.com/serde-rs/bytes#explanation
And then for the actual implementation, you'd want to use the approach in `serde_with`, which is leveraging the same technique:
https://docs.rs/serde_with/latest/serde_with/struct.Bytes.html
(serde_with is already in crate imports)

Reply:
I tried this but was miserably failing, I will attempt it again.

Reply:
Fixed.